libraries/eosiolib/contracts/include/eosio/singleton.hpp

```diff
@@ -78,6 +78,22 @@ namespace eosio
          return itr->value;
       }
 
+      const T& get_cached()
+      {
+         auto itr = _t.find(pk_value);
+         eosio::check(itr != _t.end(), "singleton does not exist");
+         return itr->value;
+      }
```

Comment:
I added it since I was there adding `get_or_null. `get` unnecessarily creates a copy, but changing it would be an api break in code like this:

```
auto& x = sing.get();
x.foo = 7;
sing.set(x, payer);
```